x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -198,3 +209,24 @@ export const getEndpointAuthzInitialState = (): EndpointAuthz => {
     canWriteEndpointExceptions: false,
   };
 };
+
+/**
+ * Duplicate logic to calculate if user has privilege to fetch Agent Policies,
+ * working only with Capabilities, in order to be able to use it e.g. in middleware.
+ *
+ * The logic works with Fleet granular privileges (`subfeaturePrivileges`) both enabled and disabled.
+ *
+ * @param capabilities Capabilities from coreStart.application
+ */
+export const canFetchPackageAndAgentPolicies = (capabilities: Capabilities): boolean => {
```

Comment:
Is there really no way to just use the regular "authz". My concern is that we now have two separate methods that calculate authz and it would be great if we could avoid that.

Reply:
i cannot say there is no way, but, i couldn't find a feasible one. i agree on your concerns, that's why i tried to emphasize in the comment that this is a duplicate logic to be maintained, but unfortunately couldn't come up with a better solution.

we can e.g. read the privileges in the ui component, and dispatch an action with them to the redux state, so there is knowledge about it, but that would be another addition to the redux part again, so i'm not sure it's worth it.

what do you think?

Reply:
yeah... don't like us having to keep adding to the Redux setup either. I guess keep it as is for now. I till have the issue assigned to me around removing redux from policy management - removed some stuff, but not all.